tests/drivers/stepper/shell/src/main.c

```diff
@@ -0,0 +1,216 @@
+	zassert_ok(err, "failed to execute shell command (err %d)", err);
+	zassert_equal(fake_stepper_set_target_position_fake.call_count, 1,
+		      "set_target_position function not called");
+	zassert_equal(fake_stepper_set_target_position_fake.arg0_val, fake_stepper_dev,
+		      "wrong device pointer");
+	zassert_equal(fake_stepper_set_target_position_fake.arg1_val, 200,
+		      "wrong target position value");
+#ifdef CONFIG_STEPPER_SHELL_ASYNC
+	zassert_not_null(fake_stepper_set_target_position_fake.arg2_val,
+			 "async poll signal not be null");
+#else
+	zassert_is_null(fake_stepper_set_target_position_fake.arg2_val,
+			"async poll signal is null");
+#endif /* CONFIG_STEPPER_SHELL_ASYNC */
+}
+
+ZTEST(stepper_shell, test_stepper_enable_constant_velocity_mode)
+{
+	const struct shell *sh = shell_backend_dummy_get_ptr();
+	int err;
+
+	err = shell_execute_cmd(sh, "stepper enable_constant_velocity_mode " FAKE_STEPPER_NAME
+				    " positive 200");
+	zassert_ok(err, "failed to execute shell command (err %d)", err);
+	zassert_equal(fake_stepper_enable_constant_velocity_mode_fake.call_count, 1,
+		      "enable_constant_velocity_mode function not called");
+	zassert_equal(fake_stepper_enable_constant_velocity_mode_fake.arg0_val, fake_stepper_dev,
+		      "wrong device pointer");
+	zassert_equal(fake_stepper_enable_constant_velocity_mode_fake.arg1_val,
+		      STEPPER_DIRECTION_POSITIVE, "wrong direction value");
+	zassert_equal(fake_stepper_enable_constant_velocity_mode_fake.arg2_val, 200,
+		      "wrong velocity value");
+}
+
+ZTEST(stepper_shell, test_stepper_enable_constant_velocity_mode_invalid_direction)
+{
+	const struct shell *sh = shell_backend_dummy_get_ptr();
+	int err;
+
+	err = shell_execute_cmd(sh, "stepper enable_constant_velocity_mode " FAKE_STEPPER_NAME
+				    " foo 200");
+	zassert_not_equal(err, 0,
+			  " executed enable_constant_velocity_mode with invalid direction value");
+}
+
+ZTEST(stepper_shell, test_stepper_info)
+{
+	const struct shell *sh = shell_backend_dummy_get_ptr();
+	int err;
+
+	err = shell_execute_cmd(sh, "stepper info " FAKE_STEPPER_NAME);
+	zassert_ok(err, "failed to execute shell command (err %d)", err);
+
+	zassert_equal(fake_stepper_is_moving_fake.call_count, 1, "is_moving function not called");
+	zassert_equal(fake_stepper_get_actual_position_fake.call_count, 1,
+		      "get_actual_position function not called");
+	zassert_equal(fake_stepper_get_micro_step_res_fake.call_count, 1,
+		      "get_micro_step_res function not called");
+}
+
+static void *stepper_shell_setup(void)
+{
+	const struct shell *sh = shell_backend_dummy_get_ptr();
+
+	/* Wait for the initialization of the shell dummy backend. */
+	WAIT_FOR(shell_ready(sh), 20000, k_msleep(1));
+	zassert_true(shell_ready(sh), "timed out waiting for dummy shell backend");
+
+	return NULL;
+}
+
+ZTEST_SUITE(stepper_shell, NULL, stepper_shell_setup, NULL, NULL, NULL);
```

Comment:
start the `TEST_SUITE` with the `ZTEST_SUITE`, improves readability imo

Reply:
Not sure I understand, you mean moving this definition to the top of the file?

Reply:
yep that's what i mean

Reply:
👍 moved it to the top